Helper/Cart.php

```diff
@@ -1719,15 +1719,15 @@ public function collectDiscounts(
         // selecting specific shipping option, so the conditional statement should also
         // check if getCouponCode is not null
         /////////////////////////////////////////////////////////////////////////////////
-        if (( $amount = abs($address->getDiscountAmount()) ) || $address->getCouponCode()) {
+        if (( $amount = abs($address->getDiscountAmount()) ) || $quote->getCouponCode()) {
```

Comment:
Why we need this change?

Reply:
Cause $address->getCouponCode() could returns empty code in some contexts, and if we look into the core code of Magento, it usually use $address->getQuote()->getCouponCode(); to get coupon code. Correct me if I miss anything